File: Frontend/src/classes/SettingsManager.ts
import User from "./User";
import {getAuth} from "firebase/auth";
import {getDatabase, ref, push, set, get, query, orderByChild, equalTo} from "firebase/database";

export default class SettingsManager {

    //Properties
    private static instance:SettingsManager | null = null;

    //Constructor
    private SettingsManager(){}
    
    //Methods
    public static getInstance():SettingsManager{
        if(this.instance == null) {
            this.instance = new SettingsManager();
        }
        return this.instance;
    }

    public async changeMail(mail:string){
        const auth = getAuth();
        const Uid = auth.currentUser?.uid;
        const db = getDatabase();

        let user = (await get(ref(db, `Users/${Uid}`))).val();
        if(user){
            await set(ref(db, `Users/${Uid}/Mail`), mail); 
        }
    }

    public async changePassword(password:string){
        const auth = getAuth();
        const Uid = auth.currentUser?.uid;
        const db = getDatabase();

        let user = (await get(ref(db, `Users/${Uid}`))).val();
        if(user){
            await set(ref(db, `Users/${Uid}/Password`), password); 
        }
        return true;
    }

    public async changePhone(phoneNum:string){
        const auth = getAuth();
        const Uid = auth.currentUser?.uid;
        const db = getDatabase();

        let user = (await get(ref(db, `Users/${Uid}`))).val();
        if(user){
            await set(ref(db, `Users/${Uid}/PhoneNum`), phoneNum); 
        }
        return true;
    }

    public async changeAddress(address:string){
        const auth = getAuth();
        const Uid = auth.currentUser?.uid;
        const db = getDatabase();

        let user = (await get(ref(db, `Users/${Uid}`))).val();
        let currentAddress = (await get(ref(db, `Users/${Uid}`))).val();
        if(user){
            await set(ref(db, `Users/${Uid}/Address`), address); 
        }
        return true;
    }

    public async changeHES(hesCode:string){
        const auth = getAuth();
        const Uid = auth.currentUser?.uid;
        const db = getDatabase();

        let user = (await get(ref(db, `Users/${Uid}`))).val();
        if(user){
            await set(ref(db, `Users/${Uid}/HesCode`), hesCode); 
        return true;
        }
    }
}

File: Backend/src/main/java/com/example/backend/SettingsManager.java
package com.example.backend;

public class SettingsManager {

    //Properties
    private static SettingsManager instance = null;

    //Constructor
    private SettingsManager(){}

    //Methods
    public static SettingsManager getInstance(){
        if(instance == null) {
            instance = new SettingsManager();
        }
        return instance;
    }

    public boolean changeMail(User user, String mail) {
        //TO DO
        return true;
    }

    public boolean changePassword(User user, String password) {
        //TO DO
        return true;
    }

    public boolean changePhone(User user, String phoneNum) {
        //TO DO
        return true;
    }

    public boolean changeAddress(User user, String address) {
        //TO DO
        return true;
    }

    public boolean changeHES(User user, String hesCode) {
        //TO DO
        return true;
    }

    public boolean changeProfilePic(User user, String picURL) {
        //TO DO
        return true;
    }


}
